spec/mongoid/extensions/set_spec.rb

```diff
@@ -31,5 +31,48 @@
     it "returns an array" do
       expect(set.mongoize).to eq([ "test" ])
     end
+
+    context "when there are mongoizable values in the container" do
+      let(:date) do
+        Date.new(2012, 1, 1)
+      end
+
+      let(:mongoized) do
+        Set.mongoize(input)
+      end
+
+      context "when the input is an array" do
+
+        let(:input) do
+          [ date ]
+        end
+
+        it "mongoizes to a set" do
+          expect(mongoized).to be_a(Array)
+        end
+
+        it "mongoizes each element in the array" do
+          expect(mongoized.first).to be_a(Time)
+        end
+
+        it "converts the elements properly" do
+          expect(mongoized.first).to eq(Time.utc(2012, 1, 1, 0, 0, 0))
+        end
+      end
+
+      context "when the input is a set" do
+        let(:input) do
+          [ date ].to_set
+        end
+
+        it "mongoizes to a set" do
+          expect(mongoized).to be_a(Array)
+        end
+
+        it "converts the elements properly" do
+          expect(mongoized.first).to eq(Time.utc(2012, 1, 1, 0, 0, 0))
+        end
+      end
+    end
```

Comment:
Is there test coverage in this file for deduplicating input array elements and for removing duplicates when mongoizers map multiple inputs to one output?

Reply:
done